config/management/operational/src/validator_config.rs

```diff
@@ -38,34 +42,47 @@ impl SetValidatorConfig {
         let client = JsonRpcClientWrapper::new(self.host);
         let sequence_number = client.sequence_number(owner_account)?;
 
-        // Retrieve the current validator / fullnode addresses and update accordingly
+        let validator_addresses = self.validator_addresses;
+        let full_node_addresses = self.full_node_addresses;
+
         let validator_config = client.validator_config(owner_account)?;
-        let validator_address = if let Some(validator_address) = self.validator_address {
-            validator_address
+        let config_validator_addresses = validator_config.validator_network_addresses;
+        let config_full_node_addresses = validator_config.full_node_network_addresses;
+
+        // TODO(philiphayes): read actual shared_val_netaddr_keys from secure storage
+        let shared_val_netaddr_keys: HashMap<_, _> = iter::once((
+            TEST_SHARED_VAL_NETADDR_KEY_VERSION,
+            TEST_SHARED_VAL_NETADDR_KEY,
+        ))
+        .collect();
+
+        // Retrieve the current validator / fullnode addresses and update accordingly
+        let validator_addresses = if !validator_addresses.is_empty() {
+            validator_addresses
         } else {
-            decode_validator_address(
-                validator_config.validator_network_address,
+            decode_validator_addresses(
```

Comment:
it might make sense to give a handle to secure storage rather than passing in the encryption keys.